### PR TITLE
add legends to series plots

### PR DESCRIPTION
**Pull request by nithishdivakar**

### CODE REVIEW

File: artist/plot.py [hunk 1]
```diff
@@ -306,7 +306,8 @@ def _clear_plot_mark_background(self, x, y, mark, markstyle):
             # make sure all background clear operations are performed first
             self.plot_series_list.insert(0, plot_series)
 
-    def _create_plot_series_object(self, x, y, xerr=[], yerr=[], options=None, legend=None):
+    def _create_plot_series_object(self, x, y, xerr=[], yerr=[], 
+                                        options=None, legend=None):
```
Comment (153957) [inline artist/plot.py]: `options=None` can still be on the first line (within 80 characters), and the second line should be aligned to the opening bracket of the function.

File: artist/templates/subplot.tex [hunk 2]
```diff
@@ -93,6 +93,9 @@
             ({{ x }}, {{ y }})
         {%- endfor %}
     };
+    {% if series.legend is not none %}
+        \addlegendentry{ {{series.legend}} }
```
Comment (153957) [inline artist/templates/subplot.tex]: Add spaces between the double brackets and the text, as is done for other variables.

**Comment by 153957** [comment]
Could you add a demo which shows this functionality?
We may need to modify the default style for the legend,
I would probably be in favour of `legend pos=outer north east`.

**Comment by davidfokkema** [comment]
Interesting idea. I'm curious how you would use it. Don't mind us, we're a little bit crazy. Up until now, we've had no use for legends; that's why the support is missing. If you look at the demo (http://davidfokkema.github.io/artist/demo_plots.pdf) you can compare figures 1 and 2. We've been leaning towards figure 2, no legend. But of course, not everyone does. Can you make a nice plot with legend and add it as a comment to this PR? Thanks!

**Comment by nithishdivakar** [comment]
@davidfokkema Most of the figures can be easily differentiated with a arrow pointing towards the plot. But legend is a requirement when the different plots are closely overlapping but still different. Dropping a small example in comment. The code for this example is available [here.](https://github.com/nithishdivakar/artist/blob/master/demo/demo_fourier.py)

plot

File: artist/plot.py (hunk 1 above)
Comment (nithishdivakar) [inline artist/plot.py]: fixed in [here](https://github.com/davidfokkema/artist/commit/4c3892c7d4ed794fede36b03fd57ced0ff2e35e1)

File: artist/templates/subplot.tex (hunk 2 above)
Comment (nithishdivakar) [inline artist/templates/subplot.tex]: fixed in [here](https://github.com/davidfokkema/artist/commit/4c3892c7d4ed794fede36b03fd57ced0ff2e35e1)

**Comment by davidfokkema** [comment]
Nice example!

**Comment by davidfokkema** [comment]
I'll first merge this and then make a commit adding this to the demo's pdf.

**Comment by davidfokkema** [comment]
Added example to the demo pdf, in 5f2401c28f3e193593eb8d84b89214e284324930. Still need to update the website, though.

**Comment by nithishdivakar** [comment]
Sure.

I'd like to say, this repository is immensely useful to me. Im working in machine learning and most of my code is already in python. So its very very easy for me to generate plots and get the tex output.

**Comment by davidfokkema** [comment]
I'm happy to hear that. I originally created Artist because I was unsatisfied with the output of other python plotting packages, like matplotlib. Those were great for interactive work, but not for inclusion in my thesis.  Nice to hear that it is useful for others as well!

**Pull request merged**